src/main/java/io/confluent/kafka/connect/datagen/DatagenConnector.java

```diff
@@ -107,4 +137,90 @@ private void validateSchemaSource(Config config) {
+  private void validateSchema(Config config) {
+    List<ConfigValue> schemaSources = config.configValues().stream()
+            .filter(v -> DatagenConnectorConfig.isExplicitlySetSchemaSource(v.name(), v.value()))
+            .collect(Collectors.toList());
+
+    Generator.Builder generatorBuilder = new Generator.Builder();
+    // don't perform this validation if multiple schema sources provided
+    if (schemaSources.size() != 1) {
+      return;
+    }
+
+    ConfigValue schemaSource = schemaSources.get(0);
+
+    switch (schemaSource.name()) {
+      case DatagenConnectorConfig.QUICKSTART_CONF:
+        if (!Quickstart.configValues.contains(((String) schemaSource.value()).toLowerCase())) {
+          schemaSource.addErrorMessage(QUICKSTART_ERR);
+          break;
+        }
+        try {
+          generator = generatorBuilder
+                  .schemaStream(getClass().getClassLoader()
+                          .getResourceAsStream(Quickstart.valueOf(((String) schemaSource.value())
+                                  .toUpperCase()).getSchemaFilename()))
+                  .build();
+        } catch (SchemaParseException | IOException e) {
+          schemaSource.addErrorMessage(QUICKSTART_SCHEMA_ERR);
+        }
+        break;
+
+      case DatagenConnectorConfig.SCHEMA_STRING_CONF:
+        try {
+          generator = generatorBuilder.schemaString((String) schemaSource.value()).build();
+        } catch (SchemaParseException e) {
+          schemaSource.addErrorMessage(SCHEMA_STRING_ERR);
+        }
+        break;
+
+      case DatagenConnectorConfig.SCHEMA_FILENAME_CONF:
+        try {
+          InputStream stream = new FileInputStream((String) schemaSource.value());
+          generator = generatorBuilder.schemaStream(stream).build();
+        } catch (FileNotFoundException e) {
+          try {
+            if (DatagenTask.class.getClassLoader()
+                    .getResource((String) schemaSource.value()) == null) {
+              schemaSource.addErrorMessage(SCHEMA_FILE_NOT_FOUND);
+              break;
+            }
+            generator = generatorBuilder.schemaStream(
+                    DatagenConnector.class.getClassLoader()
+                            .getResourceAsStream((String) schemaSource.value())).build();
+          } catch (SchemaParseException | IOException i) {
+            schemaSource.addErrorMessage(SCHEMA_FILE_ERR);
+          }
+        } catch (SchemaParseException | IOException e) {
+          schemaSource.addErrorMessage(SCHEMA_FILE_ERR);
+        }
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  private void validateSchemaKeyField(Config config) {
+    // don't perform this validation if provided schema is invalid
+    if (generator == null) {
+      return;
+    }
+    ConfigValue schemaKeyField = getConfigValue(config,
+            DatagenConnectorConfig.SCHEMA_KEYFIELD_CONF);
+    if (schemaKeyField != null && !schemaKeyField.value().equals("")) {
```

Comment:
Is there a possibility for schemaKeyField to be a bunch of whitespaces?

Reply:
Here, we're just checking if user has explicitly set the property and if not - we skip the validation. If the schemaKeyField is set to a bunch of whitespaces by the user that would be an invalid schemaKeyField and the validation will fail with the appropriate error.